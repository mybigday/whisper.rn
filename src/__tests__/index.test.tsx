import { initWhisper, releaseAllWhisper } from '..'

jest.mock('..', () => require('../../jest/mock'))

test('Mock', async () => {
  const context = await initWhisper()
  expect(context.id).toBe(1)
  expect(await context.transcribe('test.wav')).toEqual({ result: 'TEST' })
  await context.release()
  await releaseAllWhisper()
})
