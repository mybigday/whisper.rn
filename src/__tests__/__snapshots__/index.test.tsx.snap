// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock 1`] = `
[
  {
    "contextId": 1,
    "data": {
      "result": " Test",
      "segments": [
        {
          "t0": 0,
          "t1": 33,
          "text": " Test",
        },
      ],
    },
    "isCapturing": true,
    "jobId": 5000,
    "processTime": 100,
    "recordingTime": 1000,
  },
  {
    "contextId": 1,
    "data": {
      "result": " Test",
      "segments": [
        {
          "t0": 0,
          "t1": 33,
          "text": " Test",
        },
      ],
    },
    "isCapturing": false,
    "jobId": 5000,
    "processTime": 100,
    "recordingTime": 2000,
  },
  {
    "contextId": 1,
    "data": {
      "result": " Test",
      "segments": [
        {
          "t0": 0,
          "t1": 33,
          "text": " Test",
        },
      ],
    },
    "isCapturing": false,
    "jobId": 5000,
    "processTime": 100,
    "recordingTime": 2000,
  },
]
`;
